tests/test_resource_warning.py

```diff
@@ -0,0 +1,45 @@
+import asyncio
+import time
+from datetime import datetime
+
+import aiohttp
+
+
+async def _resource_warning(recwarn):
+    timeout = 3 * 60
+
+    request_interval = 60
+    request_increase = 30
+    next_request = 0
+
+    async with aiohttp.ClientSession() as client:
+
+        end = time.time() + timeout
+        while True:
+            now = time.time()
+            if now > end:
+                break
+
+            if now > next_request:
+                async with client.get('https://heise.de') as resp:
+                    status = resp.status
+                    await resp.text()
+
+                request_interval += request_increase
+                next_request = now + request_interval
+                print(f'{datetime.now()} | '
+                      f'Status: {status}, next request in {request_interval}s')
+
+            await asyncio.sleep(5)
+
+            for warn in recwarn:
+                print(f'{datetime.now()} | {warn}')
+                assert 'unclosed transport' not in str(warn.message)
```

Comment:
I think this reads better sematically:
```suggestion
            assert not any('unclosed transport' in str(w.message) for w in recwarn)
            recwarn.clear()
```